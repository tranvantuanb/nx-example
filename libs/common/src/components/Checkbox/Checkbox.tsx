import React from 'react';
import {
  Checkbox as AntdCheckbox,
  CheckboxProps as AntdCheckboxProps,
} from 'antd';
import tw, { styled } from 'twin.macro';

export type CheckboxProps = AntdCheckboxProps;

const StyledCheckbox = styled(AntdCheckbox)``;

export const Checkbox: React.FC<CheckboxProps> = (props) => {
  return <StyledCheckbox {...props} />;
};
