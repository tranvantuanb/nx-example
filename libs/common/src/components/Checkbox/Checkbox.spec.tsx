import { render } from '../../test';

import { Checkbox, CheckboxProps } from './Checkbox';

describe('Checkbox', () => {
  it('should render successfully', () => {
    const defaultProps: CheckboxProps = {};
    const { container } = render(<Checkbox {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
