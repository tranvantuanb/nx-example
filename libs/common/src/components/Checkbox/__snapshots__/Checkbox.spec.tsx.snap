// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render successfully 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper css-1iv9iew-StyledCheckbox eu8bm9l0"
  >
    <span
      class="ant-checkbox"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
  </label>
</div>
`;
